cfme/tests/integration/test_cfme_auth.py

```diff
@@ -1,90 +1,156 @@
-    user_group = None
-    if add_group == RETRIEVE_GROUP:
-        user_group = group_collection.instantiate(
-            description=data['group_name'], role="EvmRole-user",
-            user_to_lookup=data["username"], ldap_credentials=credentials)
-        if 'ext' in auth_mode:
-            user_group.add_group_from_ext_auth_lookup()
-        elif 'miq' in auth_mode:
-            user_group.add_group_from_ldap_lookup()
-        request.addfinalizer(user_group.delete)
-    elif add_group == CREATE_GROUP:
-        user_group = group_collection.create(
-            description=data['group_name'], role="EvmRole-user",
-            user_to_lookup=data["username"], ldap_credentials=credentials)
-        request.addfinalizer(user_group.delete)
-
-
-@pytest.fixture()
-def user(request, data, add_group, appliance):
-    if not data:
-        pytest.skip("No data specified for user")
-    username, password = data["username"], data["password"]
-    if 'evm_default_group' in add_group:
-        username, password = data['default_username'], data['default_password']
-        data['fullname'] = data['default_userfullname']
-    credentials = Credential(
-        principal=username,
-        secret=password,
-        verify_secret=password,
-    )
-    user_obj = appliance.collections.users.instantiate(
-        name=data['fullname'], credential=credentials
-    )
+    group_args = {
+        'description': kwargs.get('group_name'),
+        'role': kwargs.get('role')}
+    group = group_collection.create(**group_args)  # gets re-instantiated in create
+    assert group.exists
+    yield group
+
     try:
-        request.addfinalizer(user_obj.delete)
-    except CandidateNotFound:
-        logger.warning('User was not found during deletion')
-    return user_obj
+        group.delete()
+    except Exception:
+        logger.warning('Exception during group fixture cleanup')
 
 
 @pytest.mark.tier(1)
-@pytest.mark.parametrize(
-    "add_group", ['create_group', 'retrieve_group', 'evm_default_group'])
-@pytest.mark.meta(blockers=[GH('ManageIQ/integration_tests:6465',
-                               unblock=lambda auth_mode: auth_mode != 'ext_openldap')])
-def test_auth_configure(appliance, request, configure_auth, group, user, data):
+@pytest.mark.uncollectif(lambda auth_mode: auth_mode == 'amazon')  # default groups tested elsewhere
+def test_login_evm_group(appliance, auth_mode, prov_key, user_type, auth_provider, configure_auth,
+                         auth_user_data):
+    """This test checks whether a user can login while assigned a default EVM group
+        Prerequisities:
+            * ``auth_data.yaml`` file
+            * auth provider configured with user as a member of a group matching default EVM group
+        Test will configure auth and login
+    """
+    coll = appliance.collections.users
+    # get a list of (user_obj, groupname) tuples, filtering on those that have evmgroup in groupname
+    user_tuples = [
```

Comment:
**REQUIRED:** This needs to be reformatted, it's very hard to read right now and some of the indents are not 4 multiples